src/Impostor.Server/Net/Manager/CompatibilityManager.cs

```diff
@@ -0,0 +1,100 @@
+using System.Collections.Generic;
+using Impostor.Api.Innersloth;
+using Impostor.Api.Net.Manager;
+using Microsoft.Extensions.Logging;
+
+namespace Impostor.Server.Net.Manager
+{
+    internal class CompatibilityManager : ICompatibilityManager
+    {
+        private static readonly CompatData[] DefaultSupportedVersions =
+        {
+            new(GameVersion.GetVersion(2022, 11, 1), GameVersion.GetVersion(2022, 11, 1), true), // 2022.12.8
+
+            new(GameVersion.GetVersion(2022, 11, 9), GameVersion.GetVersion(2022, 11, 9), true), // 2022.12.14
+
+            new(GameVersion.GetVersion(2022, 12, 2), GameVersion.GetVersion(2022, 12, 2), true), // 2023.2.28
+
+            new(GameVersion.GetVersion(2023, 1, 11), GameVersion.GetVersion(2023, 1, 11), true), // 2023.3.28s
+            new(GameVersion.GetVersion(2023, 3, 13), GameVersion.GetVersion(2023, 1, 11), true), // 2023.3.28a
+            new(GameVersion.GetVersion(2023, 4, 21), GameVersion.GetVersion(2023, 1, 11), true), // 2023.6.13
+
+            new(GameVersion.GetVersion(2023, 5, 20), GameVersion.GetVersion(2023, 5, 20), true), // 2023.7.11
+            new(GameVersion.GetVersion(2222, 0, 0), GameVersion.GetVersion(2023, 5, 20), false), // 2023.7.11 for host-only mods
+        };
+
+        // Map from client version to compatibility group
+        private readonly Dictionary<int, int> _supportMap = new Dictionary<int, int>();
+        private readonly ILogger<CompatibilityManager> _logger;
+        private readonly bool initializationFinished;
```

Comment:
```suggestion
        private readonly bool _initializationFinished;
```

But also do we even need this? Wouldn't it be better to split `AddSupportedVersion` into a private and public versions?

Reply:
We don't need it, but considering this API is a massive footgun if used incorrectly, I wanted to print a warning if it is. 

Splitting into private/public would be a dupe and considering `private` in C# to me seems not much more than a suggestion, I don't think this would be useful